backend/fms_core/services/sample.py

```diff
@@ -304,6 +304,108 @@ def extract_sample(process: Process,
     return (sample_destination, errors, warnings)
 
 
+def pool_samples(process: Process,
+                 samples_info,
+                 pool_name,
+                 container_destination: Container,
+                 coordinates_destination,
+                 execution_date: datetime.date):
+    sample_destination = None
+    errors = []
+    warnings = []
+
+    if not process:
+        errors.append(f"Process for extraction is required.")
+    if not samples_info:
+        errors.append(f"Information must be provided about the samples to pool.")
+    if not container_destination:
+        errors.append(f"Destination container for extraction is required.")
+    if not pool_name:
+        errors.append(f"Pool Name is required.")
+    if not isinstance(execution_date, date):
+        errors.append(f"Execution date is not valid.")
+    
+    if not errors:
+        volume = sum(sample["Volume Used (uL)"] for sample in samples_info) # Calculate the total volume of the pool
+        for sample in samples_info:
+            volume_used = sample["Volume Used (uL)"]
+            sample_obj = sample["Source Sample"]
+            if volume_used is None:
+                errors.append(f"Volume used is required.")
+            else:
+                if volume_used <= 0:
+                    errors.append(f"Volume used ({volume_used}) is invalid.")
+                if sample_obj and volume_used > sample_obj.volume:
+                    errors.append(f"Volume used ({volume_used} uL) exceeds the current volume of the sample ({sample_obj.volume} uL).")
+
+            sample["Volume Ratio"] = volume_used / volume # Calculate the volume ratio of each sample in the pool
+            sample["Fractional Concentration"] = sample["Volume Ratio"] * sample_obj.concentration
+            sample_obj.volume = decimal_rounded_to_precision(sample_obj.volume - volume_used)  # Reduce the volume of source samples by the volume used
+            sample_obj.save()
+        concentration = sum(sample["Fractional Concentration"] for sample in samples_info) # Calculate pool concentration using fractional concentration
```

Comment:
Is it possible that the lab will want to normalize the pool concentration during the pooling step?

Reply:
It is more like people will want to pool after/during the normalization protocol. We'll be adding the pooling to the normalization in the next sprint. Would not make much sense to also add the normalization to the pooling.